tests/test_cli_options.py

```diff
@@ -0,0 +1,507 @@
+
+import psutil
+
+from unittest.mock import patch
+
+
+class CLISubclass(CLI):
+    """
+    Subclass of CLI to overwrite the parse method.
+    Parse takes a list of arguments instead of getting the args
+    from sys.argv
+    """
+
+    def __init__(self):
+        super().__init__()
+
+    def parse(self, parsed_commands=None):
+        args = self._parser.parse_args(parsed_commands[1:])
+        if args.subcommand is None:
+            self._parser.print_help()
+            self._parser.exit()
+        config = self._subcommand_configs[args.subcommand]
+        config.set_config_values(args)
+        return args, config
+
+
+class CLIConfigProfileStruct():
+    """
+    Struct class to hold the common variables shared between profile tests
+    """
+
+    def __init__(self):
+        #yapf: disable
+        self.args = [
+            '/usr/local/bin/model-analyzer',
+            'profile',
+            '--model-repository',
+            'foo',
+            '--profile-models',
+            'bar'
+        ]
+        #yapf: enable
+        config_profile = ConfigCommandProfile()
+        self.cli = CLISubclass()
+        self.cli.add_subcommand(cmd='profile', help='', config=config_profile)
+
+    def parse(self):
+        return self.cli.parse(self.args)
+
+
+class CLIConfigAnalyzeStruct():
+    """
+    Struct class to hold the common variables shared between analyze tests
+    """
+
+    def __init__(self):
+        #yapf: disable
+        self.args = [
+            '/usr/local/bin/model-analyzer',
+            'analyze',
+            '--analysis-models',
+            'a,b,c'
+        ]
+        #yapf: enable
+        config_analyze = ConfigCommandAnalyze()
+        self.cli = CLISubclass()
+        self.cli.add_subcommand(cmd='analyze', help='', config=config_analyze)
+
+    def parse(self):
+        return self.cli.parse(self.args)
+
+
+class CLIConfigReportStruct():
+    """
+    Struct class to hold the common variables shared between analyze tests
+    """
+
+    def __init__(self):
+        #yapf: disable
+        self.args = [
+            '/usr/local/bin/model-analyzer',
+            'report',
+            '--report-model-configs',
+            'a, b, c'
+        ]
+        #yapf: enable
+        config_report = ConfigCommandReport()
+        self.cli = CLISubclass()
+        self.cli.add_subcommand(cmd='report', help='', config=config_report)
+
+    def parse(self):
+        return self.cli.parse(self.args)
+
+
+class OptionStruct():
+
+    def __init__(self,
+                 type,
+                 stage,
+                 long_flag,
+                 short_flag=None,
+                 expected_value=None,
+                 default_value=None,
+                 expected_failing_value=None,
+                 extra_commands=None):
+
+        self.long_flag = long_flag
+        self.short_flag = short_flag
+        self.expected_value = expected_value
+        self.default_value = default_value
```

Comment:
is this the expected default value?

Reply:
This will be the expected default value that is set in config_defaults.py. I set most of those in the test.

Reply:
ok, I would recommend renaming to expected_default_value then